src/modm/platform/core/e31/startup.S

```diff
@@ -0,0 +1,58 @@
+.section .text.init.entry
+.global _entry
+_entry:
+    .cfi_startproc
+    .cfi_undefined ra
+
+.option push
+.option norelax
+    la gp, __global_pointer$
+.option pop
+
+    # init the stack pointer
+    la sp, _sp
+
+    #set up the isr for the trap interrupt
+    #TODO: set this to a proper ISR
+    la t0, lowlevel_vector
+    csrw mtvec, t0
+
+    #enable the interrupts
+    #li t0, 0x08
+    #csrw mstatus, t0
+
+    #reset and enable the rtc for use as time base
+    #move that into a hardware init c function
+    li t0, 0x10000040   #RTC Base register
+    sw zero, 0x08(t0)   #Reset the Counter Low Register
+    sw zero, 0x0C(t0)   #Reset the Counter High Register
+    li t1,   0x1000     #set the bit for the always on reg
+    sw t1,   0x00(t0)   #and store it
+
+    #init the c runtime
+    csrr a0, mhartid
+    li a1, 0
+    li a2, 0
+    call _start
```

Comment:
We'll have to do our own startup script, delegating startup to newlib is not great.
How much did you change this from the source (ie. where is this file from and what license does it have)?

Reply:
The startup script is derived from the bare metal SDK from SiFive. It is under the Apache 2.0 Licence (same for the linker script).
I will have a look into how the cortex core is starting up the C environment

Reply:
It's probably enough to jump to modm's custom startup handler instead of `_start`. The Cortex-M startup is faily simple, but you definitely need the [custom linkerscript sections](https://modm.io/reference/module/modm-platform-cortex-m/#linkerscript).